src/containers/CreatePipelineRun/CreatePipelineRun.js

```diff
@@ -14,497 +14,427 @@ limitations under the License.
+    return props[PIPELINE_REF] ? props[key] : state[key];
   }
 
-  onHelmToggleChange(value) {
-    this.checkValidation('helmPipeline', value);
-    this.setState(state => {
-      return {
-        helmPipeline: {
-          ...state.helmPipeline,
-          value
-        }
-      };
-    });
+  checkFormValidation() {
+    // Namespace, PipelineRef, Resources, and Params must all have values
+    const validNamespace = !!this.getPipelineInfo(NAMESPACE);
+    const validPipelineRef = !!this.getPipelineInfo(PIPELINE_REF);
+    const validResources =
+      !this.state.resources ||
+      Object.keys(this.state.resources).reduce(
+        (acc, name) => acc && !!this.state.resources[name],
+        true
+      );
+    const validParams =
+      !this.state.params ||
+      Object.keys(this.state.params).reduce(
+        (acc, name) => acc && !!this.state.params[name],
+        true
+      );
+    return validNamespace && validPipelineRef && validResources && validParams;
   }
 
-  onSubmit(event) {
-    event.preventDefault();
+  handleClose() {
+    this.resetForm();
+    this.props.onClose();
+  }
 
-    if (!this.checkFormValidation()) {
+  handleNamespaceChange({ selectedItem: { text } }) {
+    // Reset pipeline and service account when namespace changes
+    if (text !== this.state[NAMESPACE]) {
       this.setState({
-        submitValidationError: true
+        ...initialPipelineInfoState(),
+        [NAMESPACE]: text,
+        serviceAccount: ''
       });
-      return;
     }
+  }
+
+  handleParamChange(key, value) {
+    this.setState(state => ({
+      params: {
+        ...state.params,
+        [key]: value
+      }
+    }));
+  }
+
+  handlePipelineChange({ selectedItem: { text } }) {
+    if (text !== this.state[PIPELINE_REF]) {
+      this.setState((state, props) => {
+        const pipelineInfo = parsePipelineInfo(
+          getStore().getState(),
+          text,
+          this.getPipelineInfo(NAMESPACE, { state, props })
+        );
+        return {
+          [PIPELINE_REF]: text,
+          ...pipelineInfo,
+          resources: initialResourcesState(pipelineInfo[RESOURCE_SPECS]),
+          params: initialParamsState(pipelineInfo[PARAM_SPECS])
+        };
+      });
+    }
+  }
+
+  handleResourceChange(key, value) {
+    this.setState(state => ({
+      resources: {
+        ...state.resources,
+        [key]: value
+      }
+    }));
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    // Check form validation
+    const valid = this.checkFormValidation();
     this.setState({
-      errorMessage: '',
-      submitValidationError: false
+      validationError: !valid
     });
+    if (!valid) {
+      return;
+    }
 
-    const namespace = this.getNamespace();
-    const pipelineValue = this.getPipeline();
+    // Send API request to create PipelineRun
+    const pipelineRef = this.getPipelineInfo(PIPELINE_REF);
+    const namespace = this.getPipelineInfo(NAMESPACE);
     const payload = {
```

Comment:
constructing the payload should be the responsibility of the API
doing this in a follow up PR is fine

Reply:
Ok, I'll move this into the API call when I implement it in a follow up PR 👍